src/protocols/rdp/rdp.c

```diff
@@ -223,10 +223,53 @@ static BOOL rdp_freerdp_authenticate(freerdp* instance, char** username,
 
     rdpContext* context = instance->context;
     guac_client* client = ((rdp_freerdp_context*) context)->client;
-
-    /* Warn if connection is likely to fail due to lack of credentials */
-    guac_client_log(client, GUAC_LOG_INFO,
-            "Authentication requested but username or password not given");
+    guac_rdp_client* rdp_client = (guac_rdp_client*) client->data;
+    guac_rdp_settings* settings = rdp_client->settings;
+    char* params[4] = {};
+    int i = 0;
+    
+    if (settings->username == NULL || strcmp(settings->username, "") == 0) {
+        params[i] = "username";
+        rdp_client->rdp_credential_flags |= GUAC_RDP_CRED_FLAG_USERNAME;
+        i++;
+    }
+    
+    if (settings->password == NULL || strcmp(settings->password, "") == 0) {
+        params[i] = "password";
+        rdp_client->rdp_credential_flags |= GUAC_RDP_CRED_FLAG_PASSWORD;
+        i++;
+    }
+    
+    if (settings->domain == NULL || strcmp(settings->domain, "") == 0) {
+        params[i] = "domain";
+        rdp_client->rdp_credential_flags |= GUAC_RDP_CRED_FLAG_DOMAIN;
+        i++;
+    }
+    
+    /* NULL-terminate the array. */
+    params[i] = NULL;
+    
+    if (i > 0) {
```

Comment:
@necouchman Would it be reasonable for us to put an extra parameter here to tell what client what form to request if it doesn't have one?  That would circumvent the CONNECTION_GROUP issue

Reply:
Maybe so, but I don't think we necessarily need anything on the server-side of this, I think it can all be accomplished on the guacamole-client side (I think).  My original code had it looking up the protocol for the connection and pulling down the form manually, but I opted for this route based on some of the changes Mike had made for the hidden menu and making changes on-the-fly.  I'm sure there's a way to do this elegantly :-).

Reply:
Alrighty, I'll keep poking around on the UI!